scripts/backfill_errors.py

```diff
@@ -0,0 +1,427 @@
+    trace_id = None
+    span_id = None
+
+    try:
+        trace_idx = event.contexts_key.index("trace")
+        transaction_ctx = json.loads(event.contexts_value[trace_idx])
+    except ValueError:
+        transaction_ctx = {}
+
+    if transaction_ctx.get("trace_id", None):
+        trace_id = UUID(transaction_ctx["trace_id"])
+    if transaction_ctx.get("span_id", None):
+        span_id = int(transaction_ctx["span_id"], 16)
+
+    error = Error(
+        UUID(event.event_id),
+        event.project_id,
+        event.timestamp,
+        event.platform or "",
+        event.environment,
+        event.release,
+        event.dist,
+        ip_v4,
+        ip_v6,
+        event.user or "",
+        event.user_id,
+        event.user_name,
+        event.user_email,
+        event.sdk_name,
+        event.sdk_version,
+        event.http_method,
+        event.http_referer,
+        event.tags_key,
+        event.tags_value,
+        event.contexts_key,
+        event.contexts_value,
+        event.transaction_name or "",
+        span_id,
+        trace_id,
+        event.partition or 0,
+        event.offset or 0,
+        event.message_timestamp,
+        event.retention_days,
+        event.deleted,
+        event.group_id,
+        UUID(event.primary_hash),
+        event.received or 0,
+        event.message or "",
+        event.title or "",
+        event.culprit or "",
+        event.level,
+        event.location,
+        event.version,
+        event.type or "",
+        event.exception_stacks_type,
+        event.exception_stacks_value,
+        event.exception_stacks_mechanism_type,
+        event.exception_stacks_mechanism_handled,
+        event.exception_frames_abs_path,
+        event.exception_frames_colno,
+        event.exception_frames_filename,
+        event.exception_frames_function,
+        event.exception_frames_lineno,
+        event.exception_frames_in_app,
+        event.exception_frames_package,
+        event.exception_frames_module,
+        event.exception_frames_stack_level,
+        event.sdk_integrations,
+        event.modules_name,
+        event.modules_version,
+    )
+
+    return {
+        name: value
+        for name, value in zip(Error.field_names(), error._asdict().values())
+    }
+
+
+def backfill_errors() -> None:
+    events_storage = get_writable_storage(StorageKey.EVENTS)
+    errors_storage = get_writable_storage(StorageKey.ERRORS)
+
+    orderby_desc = "project_id DESC, timestamp DESC, event_id DESC"
+    orderby_asc = "project_id ASC, timestamp ASC, event_id ASC"
+
+    cluster = events_storage.get_cluster()
+
+    clickhouse = cluster.get_query_connection(ClickhouseClientSettings.MIGRATE)
+
+    events_table_name = events_storage.get_table_writer().get_schema().get_table_name()
+    errors_table_name = errors_storage.get_table_writer().get_schema().get_table_name()
+
+    try:
+        last_event_id, last_event_project, last_event_timestamp = clickhouse.execute(
```

Comment:
This seems the oldest event id, not the last.

Reply:
It's supposed to mean last in terms of the order of events being migrated (which is in reverse order by time), I'll try and think of a better name.